### ISSUE TITLE
CAMB fails for redshift = 0.0

### ISSUE THREAD

**Issue by rrjbca** [skypyproject/skypy#438]
**Describe the bug**
`skypy.power_spectrum.camb` fails to calculate the linear matter power spectrum for the argument `redshift = 0.0`

**To Reproduce**

```python
>>> from astropy.cosmology import Planck18
>>> import numpy as np
>>> from skypy.power_spectrum import camb
>>> wavenumber = np.logspace(-2, 1, 4)
>>> A_s, n_s = 2.2e-9, 0.76

>>> camb(wavenumber, 1e-5, Planck18, A_s, n_s)  # redshift = 1e-5
array([1.15293630e+05, 9.38033341e+03, 4.88948085e+01, 8.95805129e-02])

>>> camb(wavenumber, 0.0, Planck18, A_s, n_s)  # redshift = 0.0
 Error in dverk, x =   14165.300429599007      xend=   14165.300429599007     
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "skypy/power_spectrum/_camb.py", line 88, in camb
    pk_interp = get_matter_power_interpolator(pars,
  File "lib/python3.8/site-packages/camb/camb.py", line 351, in get_matter_power_interpolator
    results = get_results(pars)
  File "lib/python3.8/site-packages/camb/camb.py", line 35, in get_results
    res.calc_power_spectra(params)
  File "lib/python3.8/site-packages/camb/results.py", line 329, in calc_power_spectra
    self.calc_transfers(params, only_transfers=False)
  File "lib/python3.8/site-packages/camb/results.py", line 311, in calc_transfers
    config.check_global_error('calc_transfer')
  File "lib/python3.8/site-packages/camb/_config.py", line 51, in check_global_error
    raise CAMBError(reference + '%s' % err)
camb.baseconfig.CAMBError: Error in Fortran called from calc_transfer:
DVERK error
```

**Desktop (please complete the following information):**
 - OS: macOS 10.14.6
 - Python: 3.8.7
 - SkyPy: 0.4

**Suggested Solution**
I suspect the cause of this failure is that the CAMB interpolator tries to create a redshift grid between the default zmin parameter (which is zero) and `zmax=np.max(redshift)` (which is also zero). We should either specify a different configuration for the interpolator in this corner case or let the user specify the configuration via kwargs.

**Comment by ntessore** [comment]
In light of the proposed changes, instead of attacking this issue head on, we should create our own `TabulatedPowerSpectrum` interpolator and call the standard CAMB interface instead of the interpolator.

**Comment by rrjbca** [comment]
@ntessore I think we still want to address this bug immediately otherwise we risk forgetting about it and making the same mistake again. Our own power spectrum class will need to handle this corner case and the PR fix provides the necessary configuration logic and a regression test that can be reused.

**Comment by rrjbca** [comment]
Closed by #439
